### PR TITLE
Manage relative paths ( `./*` , `~/*` and `~user/*` )

### PR DESCRIPTION
**Pull request by ndeloof**
This adds support for config using `~/...` syntax for paths relative to user's HOME

### CODE REVIEW

File: cli/options.go [hunk 1]
```diff
@@ -283,3 +295,15 @@ func getAsStringList(em map[string]string) []string {
 	}
 	return m
 }
+
+// resolvePath resolves "~" in paths
+func resolvePath(path string) (string, error) {
+	if strings.HasPrefix(path, "~") {
```
Comment (rumpl) [inline cli/options.go:301]: Shouldn't this be `"~/"` since we are doing `path[2]` later

Comment (ndeloof) [reply cli/options.go:301]: should be "~"+filepath.Separator then

Comment (rumpl) [reply cli/options.go:301]: Yah, and `path[2]` should be... not sure what... I'm not fluent in windows paths but does `~\\test` mean something?

File: cli/options_test.go [hunk 2]
```diff
@@ -84,3 +85,20 @@ func TestEnvMap(t *testing.T) {
...
+	assert.NilError(t, err)
+	assert.Equal(t, path, "")
+
+	path, err = resolvePath("/tmp/foo")
+	assert.NilError(t, err)
+	assert.Equal(t, path, "/tmp/foo")
+
+	path, err = resolvePath("~/foo")
+	assert.NilError(t, err)
+	assert.Equal(t, path, filepath.Join(home, "foo"))
```
Comment (justincormack) [inline cli/options_test.go:103]: you should test plain "~" too.

File: cli/options.go (hunk 1 above)
Comment (justincormack) [reply cli/options.go]: should this work on Windows at all? This is not AFAIK a platform convention. Although WSL2 complicates things.

Comment (rumpl) [reply cli/options.go]: It would seem `cd ~` works on PowerShell

Comment (ndeloof) [reply cli/options.go:301]: Actually, I wonder we have to manage this _at all_. Isn't this a shell extension thing, that should take place BEFORE our code get ran ?

Comment (rumpl) [reply cli/options.go:301]: It's a shell extension thing yes

Comment (ndeloof) [reply cli/options.go:301]: answering myself: we need this to support use of `~/...` inside yaml as referenced files

File: loader/paths.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+)
+
+// ExpandPath resolves "~" and "." in paths to produce absolute paths
+func ExpandPath(workingDir string, path string) (string, error) {
```
Comment (rumpl) [inline loader/paths.go:27]: nit: why not name the function `ResolvePath` or simply `Resolve` since it's what this does

Comment (ndeloof) [reply loader/paths.go:27]: cause it's based on docker-compose's `expand_path`

File: loader/paths_test.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+	"gotest.tools/v3/assert"
+)
+
+func TestExpandPath(t *testing.T) {
+	workingDir := "/foo"
+	home, err := os.UserHomeDir()
+	assert.NilError(t, err)
+
+	path, err := ExpandPath(workingDir, "")
```
Comment (silvin-lubecki) [inline loader/paths_test.go:33]: nit: I guess you can use table driven test here

File: loader/paths_test.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	for _, tt := range tests {
+		t.Run(tt.path, func(t *testing.T) {
+			got, err := ExpandPath(workingDir, tt.path)
+			assert.NilError(t, err)
+			assert.Equal(t, got, tt.expected)
+		})
+	}
+}
+
+func Abs(s string) string {
```
Comment (rumpl) [inline loader/paths_test.go:90]: nit: there is no need for this function to be exported

**Pull request closed without merging**